server/server/src/discovery.rs

```diff
@@ -1,10 +1,41 @@
+use local_ip_address::list_afinet_netifas;
+use serde::Serialize;
+use service::settings::ServerSettings;
+use simple_dns::{
+    rdata::{RData, TXT},
+    Name, ResourceRecord, CLASS,
+};
+use simple_mdns::SimpleMdnsResponder;
 use std::{
     fmt::Display,
     net::{IpAddr, Ipv4Addr},
 };
 
-use local_ip_address::list_afinet_netifas;
-use service::settings::ServerSettings;
+const SERVICE_NAME: &'static str = "_omsupply._tcp.local";
+
+pub(crate) struct Discovery {
+    #[allow(dead_code)]
+    // Need to hold instance of responder after start
+    responder: SimpleMdnsResponder,
+}
+
+impl Discovery {
+    pub fn start(server_info: &ServerInfo) -> Self {
+        let response = FrontEndHost::new(server_info);
+
+        let mut responder = SimpleMdnsResponder::new(10);
+        let srv_name = Name::new_unchecked(SERVICE_NAME);
+
+        responder.add_resource(ResourceRecord::new(
```

Comment:
Ideally we would change ServerInfo if ip was changed for some reason (i think this is sufficient for now)